src/main/java/com/zaxxer/hikari/pool/HikariPool.java

```diff
@@ -222,13 +222,15 @@ public final synchronized void shutdown() throws InterruptedException
             houseKeepingExecutorService.awaitTermination(5L, TimeUnit.SECONDS);
          }
 
-         final ExecutorService assassinExecutor = createThreadPoolExecutor(config.getMaximumPoolSize(), "Hikari connection assassin",
+         connectionBag.close();
+
+         final ExecutorService assassinExecutor = createThreadPoolExecutor(config.getMaximumPoolSize(), "Hikari connection assassin (pool " + poolName + ")",
                                                                            config.getThreadFactory(), new ThreadPoolExecutor.CallerRunsPolicy());
          try {
             final long start = clockSource.currentTime();
             do {
-               softEvictConnections();
                abortActiveConnections(assassinExecutor);
+               softEvictConnections();
```

Comment:
Why were these two switched?  We should definitely be trying soft eviction before abort.